docs/guides/search-indexing.rst

```diff
@@ -0,0 +1,322 @@
+
+.. code-block:: html
+   :emphasize-lines: 3
+
+   <h2>
+      Section title
+      <a class="headerlink" title="Permalink to this headline">¶</a>
+   </h2>
+
+- **Code blocks**: If a code block contains line numbers,
+  wrap them in a ``linenos`` or ``lineno`` class,
+  so they won't be indexed as part of the code.
+
+.. code-block:: html
+   :emphasize-lines: 3-7
+
+   <table class="highlighttable">
+      <tr>
+         <td class="linenos">
+            <div class="linenodiv">
+               <pre>1 2 3</pre>
+            </div>
+         </td>
+
+         <td class="code">
+            <div class="highlight">
+               <pre>First line
+   Second line
+   Third line</pre>
+            </div>
+         </td>
+      </tr>
+   </table>
+
+Overriding the default search
+-----------------------------
+
+Static sites usually have their own static index,
+and search results are retrieved via JavaScript.
+In order for Read the Docs to override the default search as expected,
+themes from the supported generators must follow these conventions.
+
+.. note::
+
+   Read the Docs will fallback to the original search in case of an error or no results.
+
+Sphinx
+~~~~~~
+
+Sphinx's basic theme provides the `static/searchtools.js`_ file,
+which initializes search with the ``Search.init()`` method.
+Read the Docs overrides the ``Search.query`` method and makes use of ``Search.output.append`` to add the results.
+A simplified example looks like this:
+
+.. code-block:: js
+
+   var original_search = Search.query;
+
+   function search_override(query) {
+      var results = fetch_resuls(query);
+      if (results) {
+         for (var i = 0; i < results.length; i += 1) {
+            var result = process_result(results[i]);
+            Search.output.append(result);
+         }
+      } else {
+         original_search(query);
+      }
+   }
+
+   Search.query = search_override;
+
+   $(document).ready(function() {
+      Search.init();
+   });
+
+Highlights from results will be in a ``span`` tag with the ``highlighted`` class
+(``This is a <span class="highlighted">result</span>``).
+If your theme works with the search from the basic theme, it will work with Read the Docs' SSS.
+
+.. _`static/searchtools.js`: https://github.com/sphinx-doc/sphinx/blob/275d9/sphinx/themes/basic/static/searchtools.js
+
+MkDocs
+~~~~~~
+
+Search on MkDocs is provided by the `search plugin`_, which is included (and activated) by default in MkDocs.
+The js part of this plugin is included in the `templates/search/main.js`_ file,
+which subscribes to the ``keyup`` event of the ``#mkdocs-search-query`` element
+to call the ``doSearch`` function (available on MkDocs >= 1.x) on every key press.
+
+Read the Docs overrides the ``initSearch`` and ``doSearch`` functions
+to subscribe to the ``keyup`` event of the ``#mkdocs-search-query`` element,
+and puts the results into the ``#mkdocs-search-results`` element.
+A simplified example looks like this:
+
+.. code-block:: js
```

Comment:
This is a really complex code example in our docs, and seems likely to get out of date. It's super useful though, so I think it's fine, I just worry about it being super specific vs. something a bit more generic.